File: back-fitzhi/src/main/java/com/fitzhi/data/internal/Constellation.java
package com.fitzhi.data.internal;

import java.io.Serializable;

import lombok.AllArgsConstructor;
import lombok.Data;

/** FRVI8611
 * <p>
 * This object is called &laquo; <strong>Constellation</strong> &raquo; 
 * because its representation on the Angular front-end is a serie of &#x2605; stars.
 * </p>
 * <p>A constellation in fact, represents the number of units of skills registered for a skill.
 * Each &#x2605; represents a &laquo; unit of skill &raquo;. 
 * E.g., a Java expert (5 &#x2605;) brings 5 ​​units to the count, when  one novice developer (1 &#x2605;) brings one.
 * </p>
 * <p>
 * Every month, a constellation will be saved in Fitzhì.
 * </p>
 */
@AllArgsConstructor(staticName = "of")
public @Data class Constellation implements Serializable {

	/**
	 * Identifier of the skill corresponding to the constellation.
	 */
	int idSkill;

	/**
	 * Number of internal stars counted for this skill.
	 */
	private int starsNumber;
	
	/**
	 * Number of internal AND EXTERNAL stars counted for this skill.
	 */
	private int starsNumberWithExternal;
}

File: front-fitzhi/src/app/dashboard/starfield/data/constellation.ts
/**
 * The class represents a constellation of skills levels.
 * A constellation is a group of <span>&#x2605;</span> to be drawn
 * in the starfield component.
 */
export class Constellation {

	/**
	 * @param idSkill the skill identifier
	 * @param count The number of stars counted for this constallation
	 * @param color the color of each star
	 * @param backgroundColor the background color of the HTML element containing each star
	 */
	constructor(
		public idSkill: number,
		public count: number,
		public color?: string,
		public backgroundColor?: string) {}

}
